biosynseq/generate.py

```diff
@@ -0,0 +1,158 @@
+"""Generate synthetic sequences and embeddings given model weights."""
+import logging
+import os
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict, List
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+from gene_transformer.config import ModelSettings
+from gene_transformer.model import (
+    LoadDeepSpeedStrategy,
+    LoadPTCheckpointStrategy,
+    ModelLoadStrategy,
+    inference,
+)
+from gene_transformer.utils import non_redundant_generation, seqs_to_fasta
+
+logger = logging.getLogger(__name__)
+
+
+def generate_fasta(
+    model_strategy: ModelLoadStrategy, fasta_path: str, num_seqs: int
+) -> Dict[str, List[str]]:
+    """Given pt file or deepspeed directory, output generated sequences' fasta file.
+
+    Parameters
+    ----------
+    model_strategy : ModelLoadStrategy
+        Model used to generate sequences, depending on whether a pt file or a deepspeed directory is given.
+    fasta_path : str
+        Path to save fasta sequences.
+    num_seqs : int
+        Number of non-redundant sequences to generate.
+
+    Returns
+    -------
+    Dict[str, List[str]]
+        Unique generated sequences.
+    """
+
+    # obtain model
+    model = model_strategy.get_model()
+    model.cuda()
+    # generate non-redundant sequences
+    results = non_redundant_generation(
+        model=model.model, tokenizer=model.tokenizer, num_seqs=num_seqs
+    )
+    results["unique_seqs"] = list(results["unique_seqs"])
```

Comment:
Lets move this line to be right above the return statement, just incase seqs_to_fasta depends on the object being a set